core/common/src/main/java/alluxio/PropertyKey.java

```diff
@@ -1273,4 +1303,26 @@ public String getDefaultValue() {
     Object value = DEFAULT_VALUES.get(this);
     return value != null ? value.toString() : null;
   }
+
+  /**
+   * @param name the name of a property key
+   * @return if this property key is deprecated
+   */
+  public static boolean isDeprecated(String name) {
+    try {
+      PropertyKey key = new PropertyKey(name);
+      Class c = key.getClass();
+      Field field = c.getDeclaredField(name);
+      Annotation[] annotations = field.getDeclaredAnnotations();
+      for (Annotation anno : annotations) {
+        if (anno instanceof Deprecated) {
+          return true;
+        }
+      }
+      return false;
+    } catch (NoSuchFieldException e) {
+      System.err.println(String.format("No Such Field Exception: %s", e.getMessage()));
```

Comment:
typically we don't print to std out or std err. 
I think we are ok to ignore here.